Cython/Build/Dependencies.py

```diff
@@ -392,13 +392,17 @@ def parse_code(start: cython.Py_ssize_t, in_fstring: cython.bint = False) -> cyt
 # We need to allow spaces to allow for conditional compilation like
 # IF ...:
 #     cimport ...
-dependency_regex = re.compile(r"(?:^\s*from +([0-9a-zA-Z_.]+) +cimport)|"
-                              r"(?:^\s*cimport +([0-9a-zA-Z_.]+(?: *, *[0-9a-zA-Z_.]+)*))|"
-                              r"(?:^\s*cdef +extern +from +['\"]([^'\"]+)['\"])|"
-                              r"(?:^\s*include +['\"]([^'\"]+)['\"])", re.M)
+dependency_regex = re.compile(
+    r"(?:^\s*from +cython\.cimports\.([\w.]+) +c?import)|"
```

Comment:
I think some of the ` +` should be `\s+`. Python definitely seems to accept tabs in the middle of imports (e.g. `exec("from\tmath import\t\tsin")` works for me)

Reply:
We're parsing across lines here, I guess that was the reason why the code didn't use `\s` (which includes newline). But I don't think a newline can realistically appear before the import name list in the end. I'll add `\s` at least in a few places.

Reply:
Ah I forgot about newlines. Looking at https://docs.python.org/3/reference/lexical_analysis.html it should probably be `[ \t\f]` then.